Job Board MVP

klin2020 opened the pull request:
## Summary

To reduce mass-emails sent on CoP listservs and promote technology-forward job postings, DG seeks to create a public job board. 

The Job Board exists as a separate tab on the website. The contents of the job board are manually copy/pasted into the `job_board.yml` file. Job listings that have expired are automatically removed from the job board. Guides to manage the job board, including adding and removing job posts, are available for review and reference.

<img width="1338" alt="Screenshot 2024-07-26 at 3 21 16 PM" src="https://github.com/user-attachments/assets/cfe175c6-5400-481d-a008-9454670115ac">

## Preview

[Link to Preview](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/kl-job-board-mvp/job-board/)

<!--
⚠️ Significant visual changes require submitting previous design to wayback machine.
-->

## Solution
The goal of this project was to develop a job board that used minimal dependencies and was developed to be as lightweight as possible 

The job board is a Hugo markdown page connected to a `list.html` file and a shortcode called `job-board.html`. To edit the formatting of the job board postings, refer to the shortcode `job-board.html`. All styling for the job board can be found in `_card-job.scss` To edit the job postings shown on the job board, you must edit the YAML content on `job_board.yml`

## Limitations
Currently, content managers must review listserv activity for new job board postings. Future iterations of the job board seek to eventually migrate all job posting activity away from CoP listservs. Methods to divert job posting requests from listserv to other avenues (i.e. zendesk emails, forms) have been considered.

The current iteration of the job board also requires content managers to make a PR every time an update to the job board needs to be made. More detailed information on how to manage the job board is available. 

If there are no job posts on the job board, text appears to indicate that the empty job board page is intentional. When there are many job postings, however, there is currently no pagination formatting. This is because we currently do not have pagination formatting for non-page elements.


<!--
It can be helpful if we understand:
1. What the solution is,
2. Why this approach was chosen,
3. How you implemented the change, and
4. Possible limitations of this approach and alternate solution paths.
-->


## How To Test

1. First Step
2. Second Step
3. Third Step

<!--
For PRs that include dependency updates, uncomment this section and
include a list of the changed dependencies and version numbers.
-->

<!--
## Dependency updates

| Dependency name              | Previous version | New version |
| ---------------------------- | :--------------: | :---------: |
| [Updated dependency example] |     [1.0.0]      |   [1.0.1]   |
| [New dependency example]     |        --        |   [3.0.1]   |
| [Removed dependency example] |     [2.10.2]     |     --      |
-->

<!--
Before opening this PR, make sure you’ve done whichever of these applies to you:
- [ ] Branch is up-to-date and includes latest from `main`
- [ ] PR has correct labels
- [ ] A11y testing (voice over testing, meets WCAG, run axe tools)
-->

nick-mon1 commented on themes/digital.gov/layouts/shortcodes/job-board.html:
### Improvement

Short codes are meant for content users to add to the content. Hugo best practice would be to create a `job-board/list.html` instead.

nick-mon1 commented on themes/digital.gov/layouts/shortcodes/job-board.html:
```diff
@@ -0,0 +1,72 @@
+{{ range .Site.Data.job_board }}
+  {{ if ge .close_date $now }}
+    {{ $anyActivePost = true }}
+    <article class="card card-job" aria-labelledby="{{ .title }}">
```
### Suggestions/Improvements

<img width="798" alt="Screenshot 2024-08-01 at 5 17 34 PM" src="https://github.com/user-attachments/assets/c56a3907-64e6-4d3d-b658-02ea44f841a8">

<img width="752" alt="Screenshot 2024-08-01 at 5 17 25 PM" src="https://github.com/user-attachments/assets/5e72321c-5de4-4be0-9346-9cc823364a40">

nick-mon1 reviewed (changes requested):
Excellent work in putting this together, I've noted some minor updates and best practices. Great work @klin2020

nick-mon1 commented on themes/digital.gov/layouts/shortcodes/job-board.html:
```diff
@@ -0,0 +1,72 @@
+        </div>
+      </div>
+      <div class="card-job__apply">
+        <a
```
We should set the `aria-label` so screen readers will read `Apply for job title goes here`.

```html
aria-label="Apply for  {{ .title }}"
```

RileySeaburg reviewed (approved):
Really great job!

klin2020 commented:
Hi @nick-mon1 - Each change should live in its own commit, but here is a summary of the changes made based off of your suggestions. 

- BEMify `card-job__content` into `card-job__content-header`
- Update Update type functions into more common functions `font-size` `line-height` and `font-weight`
- BEMify `card-job__meta` into `card-job__meta-date` and `card-job__meta-topics`
- Moved short code `job-board.html` code to `job-board/list.html` to align with Hugo best practice
- Bolded `department` text to improve hierarchy
- Moved CoP topics to new line
- Removed white space that occurs in narrow media window
- Set job title as link to job posting
- Set `aria-label` so screen readers will read `Apply for job title goes here.`

Other notes
- Did not change “Apply” button location on each job card to keep with formatting of other similar job boards (ex: https://www.hhs.gov/careers/job-search)

Preview of page: https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/kl-job-board-mvp/job-board/

klin2020 reviewed (commented):
All suggested changes by Nick have been made or addressed

RileySeaburg reviewed (dismissed):
Updates look great as well., thank you for making it more user friendly.

RileySeaburg commented:
@nick-mon1 you're listed on here as requesting changes and pending review. 

Can we please get this approved?

RileySeaburg reviewed (changes requested):
Can we get a readme into each of these folders to just explain what's going on. 

Can you also update the main readme to link to those readme's that you create. 

This will allow users in the future to understand how to maintain this.

@klin2020

RileySeaburg reviewed (approved):
Thank you and great job!

RileySeaburg commented on themes/digital.gov/layouts/job-board/list.html:
```diff
@@ -0,0 +1,102 @@
+              <h2 class="deck">{{- .Params.deck | markdownify -}}</h2>
+            {{- end -}}
+            <br />
+            <!-- <div id="submit_job_postings">To request a job posting to be published, email us at <a href = "mailto: digitalgov@gsa.gov">digitalgov@gsa.gov</a>. Include "Job Posting" in the subject line, the job link, a plain text position description (optional), and the related CoP(s) to the posting.</div> -->
+            <!--<div> {{ partial "core/usa-button.html" (dict "href" .RelPermalink "text" "Submit Job Posting") }} </div>-->
```
Good catch, please remove these!

klin2020 replied:
These may be used later in future implementations of the job board. Either when we divert job posting emails away from CoP listservs and to the digitalgov email, or when we create an online form for job postings. Keeping there for future reference!

klin2020 replied:
I did not see Riley's comment- I can remove them and just indicate in the technical document that these options may be taken in future directions for the job board
